core/arch/arm/kernel/boot.c

```diff
@@ -82,7 +82,9 @@ struct dt_descriptor {
 #endif
 };
 
+#ifdef CFG_EXTERNAL_DT
 static struct dt_descriptor external_dt __nex_bss;
```

Comment:
I would suggest to remove this condition and add `__maybe_unused` attribute:
`static struct dt_descriptor external_dt __nex_bss __maybe_unused;`